src/bot/mod.rs

```diff
@@ -89,4 +92,112 @@ impl Bot {
+                      ("reply_to_message_id", reply_to_message_id),
+                      ("reply_markup", reply_markup)];
+        self.post_message(&path, &params)
+    }
+
+    /// API call which will reply to a message directed to your bot.
+    pub fn reply_to_message(&self, update: &Update, text: &str) -> Result<Message> {
+        let message = update.clone().message.unwrap();
+        let message_id = message.message_id;
+        let chat_id = message.chat.id;
+        self.send_message(&chat_id, text, None, None, None, Some(&message_id), None)
+    }
+
+    /// API call which will forward a message.
+    pub fn forward_messge(&self,
+                          update: &Update,
+                          chat_id: &i64,
+                          disable_notification: Option<&bool>)
+                          -> Result<Message> {
+        let message = update.clone().message.unwrap();
+        let chat_id: &str = &chat_id.to_string();
+        let from_chat_id: &str = &message.chat.id.to_string();
+        let message_id: &str = &message.message_id.to_string();
+        let disable_notification: &str = &disable_notification.unwrap_or(&false).to_string();
+        let path = ["forwardMessage"];
+        let params = [("chat_id", chat_id),
+                      ("from_chat_id", from_chat_id),
+                      ("disable_notification", disable_notification),
+                      ("message_id", message_id)];
+        self.post_message(&path, &params)
+    }
+
+    /// API call which will send the requested chat action to your bot.
+    pub fn send_chat_action(&self, chat_id: &i64, action: &ChatAction) -> Result<bool> {
+        let chat_id: &str = &chat_id.to_string();
+        let action = &get_chat_action(action);
+        let path = ["sendChatAction"];
+        let params = [("chat_id", chat_id),
+                                 ("action", action)];
+
+        let url = ::construct_api_url(&self.bot_url, &path);
+        let mut data = self.client.post(&url).form(&params).send()?;
+        let rjson: Value = check_for_error(data.json()?)?;
+        let result = rjson.get("result").unwrap().as_bool().unwrap();
+        Ok(result)
+    }
+
+    /// API call which will send the given contact.
+    pub fn send_contact(&self,
+                        chat_id: &i64,
+                        contact: &Contact,
+                        disable_notification: Option<&bool>,
+                        reply_to_message_id: Option<&i32>,
+                        reply_markup: Option<&InlineKeyboardMarkup>) -> Result<Message> {
+
+        let chat_id: &str = &chat_id.to_string();
+        let disable_notification: &str = &disable_notification.unwrap_or(&false).to_string();
+        let reply_to_message_id: &str = &reply_to_message_id.map(|i| i.to_string())
+            .unwrap_or("None".to_string());
+        let reply_markup = &reply_markup.map(|r| serde_json::to_string(r).unwrap_or("".to_string())).unwrap_or("".to_string());
```

Comment:
This is very unreadable, could you try to cut it down? Maybe try something like:

```rust
reply_markup.and_then(|r| serde_json::to_string(r).ok())).unwrap_or("".to_string());
```

I haven't tested it.